CMakeLists.txt

```diff
@@ -25,9 +24,14 @@ set(CHRONOLOG_PACKAGE_TARNAME "${CHRONOLOG_PACKAGE}")
 #------------------------------------------------------------------------------
 # Setup install and output Directories
 #------------------------------------------------------------------------------
-#if(NOT CMAKE_INSTALL_PREFIX)
-set(CMAKE_INSTALL_PREFIX /home/$ENV{USER}/chronolog)
-#endif()
+# Validate CMAKE_BUILD_TYPE
+if(NOT (CMAKE_BUILD_TYPE STREQUAL "Debug" OR CMAKE_BUILD_TYPE STREQUAL "Release"))
+    message(FATAL_ERROR "CMAKE_BUILD_TYPE must be either Debug or Release")
+endif()
+
+set(INSTALL_DIR "/home/$ENV{USER}/chronolog" CACHE PATH "Installation directory")
+set(CMAKE_INSTALL_PREFIX "${INSTALL_DIR}/${CMAKE_BUILD_TYPE}")
+
 if(NOT CHRONOLOG_INSTALL_BIN_DIR)
```

Comment:
I would just set the directories in lines 35 - 45 without checking if they already have values,  safer this way